src/matcher.py

```diff
@@ -13,58 +16,118 @@
+    else:
+        if settings.s3_upload_location is not None:
+            s3_client.copy(
+                settings.s3_source_location.as_dict(),
+                settings.s3_upload_location.bucket,
+                settings.s3_upload_location.key
+            )
+
+    logger.info("Key %s processed", settings.s3_source_location.key)
+
+    return antivirus_results_dict(settings.file_id, matched_antivirus_rules, handler_trigger_time)
 
-    match = rules.match(f"{root_path}/{original_path}")
-    results = [x.rule for x in match]
 
-    original_s3_key = f"{user_id}/{consignment_id}/{file_id}"
-    copy_s3_key = f"{consignment_id}/{file_id}"
+@dataclass(frozen=True)
+class S3Location:
+    bucket: str
+    key: str
 
-    copy_source = {
-        "Bucket": dirty_bucket_name,
-        "Key": original_s3_key
+    def as_dict(self) -> object: return {
+        "Bucket": self.bucket,
+        "Key": self.key
     }
 
-    if len(results) > 0:
-        s3_client.copy(
-            copy_source,
-            "tdr-upload-files-quarantine-" + environment,
-            copy_s3_key
+
+@dataclass(frozen=True)
+class VirusCheckSettings:
+    file_id: str
+    s3_source_location: S3Location
+    s3_quarantine_location: Optional[S3Location]
+    s3_upload_location: Optional[S3Location]
+    local_download_location: str
+
+
+class ScanType(Enum):
+    metadata = 1
+    consignment = 2
+
+
+def build_settings(event: dict) -> VirusCheckSettings:
+    scan_type = ScanType[event.get("scanType", "consignment")]
+    consignment_id = event["consignmentId"]
+    file_id = event["fileId"]
+
+    environment = os.environ["ENVIRONMENT"]
+    efs_root_location = os.environ["ROOT_DIRECTORY"]
+    root_path = f"{efs_root_location}/{consignment_id}"
+    if scan_type == ScanType.consignment:
+        user_id = event["userId"]
+        original_path = event["originalPath"]
+        return VirusCheckSettings(
+            file_id=file_id,
+            s3_source_location=S3Location(
+                bucket="tdr-upload-files-cloudfront-dirty-" + environment,
+                key=f"{user_id}/{consignment_id}/{file_id}"
+            ),
+            s3_quarantine_location=S3Location(
+                bucket="tdr-upload-files-quarantine-" + environment,
+                key=f"{consignment_id}/{file_id}"
+            ),
+            s3_upload_location=S3Location(
+                bucket="tdr-upload-files-" + environment,
+                key=f"{consignment_id}/{file_id}"
+            ),
+            local_download_location=f"{root_path}/{original_path}"
         )
-    else:
-        s3_client.copy(copy_source, "tdr-upload-files-" + environment, copy_s3_key)
+    elif scan_type == ScanType.metadata:
+        return VirusCheckSettings(
+            file_id=file_id,
+            s3_source_location=S3Location(
+                bucket="tdr-draft-metadata-" + environment,
+                key=f"{consignment_id}/{file_id}"
+            ),
+            s3_quarantine_location=None,
```

Comment:
If a virus is found the metadata file should still go to the quarantine bucket. Will need a specific key to distinguish it from any other files from the consignment that go to the quarantine bucket, eg: `{consignment_id}/metadata/{file_id}`